File: code-monkey/src/tools/ca/ca_tool.py
from typing import Any
from langchain_core.tools import BaseTool
from deps import ASTParser


class CATool(BaseTool):
    parser: ASTParser

    def __init__(self, **kwargs: Any):
        if "parser" not in kwargs:
            kwargs["parser"] = ASTParser()
        super().__init__(**kwargs)

File: code-monkey-ts/tools/ca/ca_tool.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CATool = void 0;
const tools_1 = require("@langchain/core/tools");
const deps_1 = require("../deps");
class CATool extends tools_1.BaseTool {
    constructor(options = {}) {
        super();
        this.parser = options.parser || new deps_1.ASTParser();
    }
}
exports.CATool = CATool;
